demo/include/demo/types.hpp

```diff
@@ -84,15 +84,22 @@ struct Command {
 };
 
 struct Move {
-    Direction direction;
-    Position deltaPosition;
+    explicit Move(const Direction& direction)
+            : direction{direction}, deltaPosition{directionDeltaPositionMap().at(direction)} {
+    }
 
     static std::vector<Move> possibleMoves() {
+        return {Move{Direction::UP}, Move{Direction::DOWN}, Move{Direction::LEFT}, Move{Direction::RIGHT}};
+    }
```

Comment:
I did wonder again, why we did this. After some digging into our previous discussion in https://github.com/KIT-MRT/arbitration_graphs/pull/9#discussion_r1617816733 and trying different solutions, I'd still stay with this approach…

```suggestion
    // It would be nicer to have a static member here, but as we're self-referencing Move here, that doesn't work (incomplete type).
    // Using a static member function returning this list is still nicer than a global constant variable
    static std::vector<Move> possibleMoves() {
        return {Move{Direction::UP}, Move{Direction::DOWN}, Move{Direction::LEFT}, Move{Direction::RIGHT}};
    }
```